ios/Capacitor/Capacitor/WebViewAssetHandler.swift

```diff
@@ -160,41 +166,45 @@ open class WebViewAssetHandler: NSObject, WKURLSchemeHandler {
-                var newHeaders: [AnyHashable: Any] = [:]
+            if self.pendingTasks.contains(urlSchemeTask.hash) {
+                if let response = response as? HTTPURLResponse {
+                    let existingHeaders = response.allHeaderFields
+                    var newHeaders: [AnyHashable: Any] = [:]
 
-                // if using live reload, then set CORS headers
-                if self.isUsingLiveReload(url) {
-                    newHeaders = [
-                        "Access-Control-Allow-Origin": self.serverUrl?.absoluteString ?? "",
-                        "Access-Control-Allow-Methods": "GET, HEAD, OPTIONS, TRACE"
-                    ]
-                }
+                    // if using live reload, then set CORS headers
+                    if self.isUsingLiveReload(url) {
+                        newHeaders = [
+                            "Access-Control-Allow-Origin": self.serverUrl?.absoluteString ?? "",
+                            "Access-Control-Allow-Methods": "GET, HEAD, OPTIONS, TRACE"
+                        ]
+                    }
 
-                if let mergedHeaders = existingHeaders.merging(newHeaders, uniquingKeysWith: { (_, newHeaders) in newHeaders }) as? [String: String] {
+                    if let mergedHeaders = existingHeaders.merging(newHeaders, uniquingKeysWith: { (_, newHeaders) in newHeaders }) as? [String: String] {
 
-                    if let responseUrl = response.url {
-                        if let modifiedResponse = HTTPURLResponse(
-                            url: responseUrl,
-                            statusCode: response.statusCode,
-                            httpVersion: nil,
-                            headerFields: mergedHeaders
-                        ) {
-                            urlSchemeTask.didReceive(modifiedResponse)
+                        if let responseUrl = response.url {
+                            if let modifiedResponse = HTTPURLResponse(
+                                url: responseUrl,
+                                statusCode: response.statusCode,
+                                httpVersion: nil,
+                                headerFields: mergedHeaders
+                            ) {
+                                urlSchemeTask.didReceive(modifiedResponse)
+                            }
                         }
-                    }
 
-                    if let data = data {
-                        urlSchemeTask.didReceive(data)
+                        if let data = data {
+                            urlSchemeTask.didReceive(data)
+                        }
                     }
                 }
+                urlSchemeTask.didFinish()
+                self.pendingTasks.removeAll(where: { $0 == urlSchemeTask.hash })
+                return
             }
-            urlSchemeTask.didFinish()
-            return
         }
 
         task.resume()
+        task.cancel()
```

Comment:
Was this left on accident?

Reply:
yep, I tested canceling the task programmatically and removed it, but then added some prints to do some more tests and when I undid the print changes I accidentally added this again, good catch!